src/OpenSage.Game/StatePersister.cs

```diff
@@ -367,15 +369,22 @@ public override void SkipUnknownBytes(int numBytes)
         public override void LogToSegmentEnd()
         {
             var segment = Segments.Pop();
-            Console.WriteLine(segment.Name);
-            for (var i = _binaryReader.BaseStream.Position; i < segment.End; i++)
+            LogBytes(segment.End - _binaryReader.BaseStream.Position, segment.Name);
+            Segments.Push(segment);
+        }
+
+
+        /// <summary>
+        /// Only to be used for local debugging and should not be present in committed code.
+        /// </summary>
+        public override void LogBytes(long numBytes, string source = "")
```

Comment:
`LogToSegmentEnd()` only works when we have a segment - which we don't always. It works great for Behaviors, but not for nested objects. This is a more versatile logging method which allows for dumping data of specific objects.